scan/main.py

```diff
@@ -22,28 +27,37 @@
 if not openai_api_key:
     raise ValueError("The OPENAI_API_KEY environment variable is not set.")
 
+client = OpenAI(api_key=openai_api_key)
+
+# Define a prompt template for general queries
+general_prompt_template = PromptTemplate(
+    input_variables=["query"],
+    template="You are a helpful AI assistant. Provide a friendly and informative response to the following query: {query}"
+)
 
 class CustomCrew:
     def __init__(self, topic):
         self.topic = topic
-        self.llm = OpenAIWrapper(api_key=openai_api_key)  # Initialize the LLM wrapper
-        self.agents = PFCAgents(llm=self.llm.llm, topic=self.topic)  # Pass topic to agents
-        self.tasks = PFCTasks(self.agents)  # Initialize tasks with agents
-
-    def run(self):
-        decision_task = self.tasks.complex_decision_making_task(self.topic)
-        emotional_task = self.tasks.emotional_risk_assessment_task(self.topic)
-        reward_task = self.tasks.reward_evaluation_task(self.topic)
-        conflict_task = self.tasks.conflict_resolution_task(self.topic)
-        social_task = self.tasks.social_cognition_task(self.topic)
+        self.llm = OpenAIWrapper(api_key=openai_api_key)
+        self.agents = PFCAgents(llm=self.llm.llm, topic=self.topic)
+        self.tasks = PFCTasks(self.agents)
+
+    def run(self, conversation_history):
+        context = "\n".join(conversation_history[-5:])  # Use last 5 exchanges for context
+        
+        decision_task = self.tasks.complex_decision_making_task(self.topic, context)
+        emotional_task = self.tasks.emotional_risk_assessment_task(self.topic, context)
+        reward_task = self.tasks.reward_evaluation_task(self.topic, context)
+        conflict_task = self.tasks.conflict_resolution_task(self.topic, context)
+        social_task = self.tasks.social_cognition_task(self.topic, context)
```

Comment:
I think I would make `conversation_history` part of the class rather than store it outside and passing it in since `CustomCrew` is the only place it is used.

```
class CustomCrew:
    def __init__(self, topic):
        self.topic = topic
        self.llm = OpenAIWrapper(api_key=openai_api_key)
        self.agents = PFCAgents(llm=self.llm.llm, topic=self.topic)
        self.tasks = PFCTasks(self.agents)
        self.conversation_history = []

    def run(self, conversation_history):
        self.conversation_history.append(conversation_history)
        context = "\n".join(self.conversation_history[-5:])  # Use last 5 exchanges for context

        decision_task = self.tasks.complex_decision_making_task(self.topic, context)
        emotional_task = self.tasks.emotional_risk_assessment_task(self.topic, context)
        reward_task = self.tasks.reward_evaluation_task(self.topic, context)
        conflict_task = self.tasks.conflict_resolution_task(self.topic, context)
        social_task = self.tasks.social_cognition_task(self.topic, context)
```